Expensify/App
# Make TextLink component accessible

cc @tgolen 

### Details
Introduced ExternalLink component to make it easier to create accessibility friendly external links. I've also separated nested Text components with ExternalLink into separate Text and ExternalLink components, so Safari and iOS can recognise links.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify.cash/issues/3100

### Tests
* Verified terms of service and privacy policy links are accessible via keyboard from Safari and Chrome on MacOS.

### QA Steps
* Launch the app/webapp
* Prerequisite: Turn ON the Screen reader.
* Mobile: Navigate using Swipe gesture(From Left to Right) and verily that terms of conditions and privacy policy links are accessible and clickable.
* Web Safari: Navigate the page using arrow keys then verify that terms of conditions and privacy policy links are accessible and clickable.
* Web Chrome: Navigate the page using tab key then verify that terms of conditions and privacy policy links are accessible and clickable.

### Tested On

- [X] Web (Safari & Chrome on MacOS)

### Screenshots

#### Web

https://user-images.githubusercontent.com/3853003/120563396-5f216400-c400-11eb-88c9-97c0c4fa6055.mov

#### Desktop

https://user-images.githubusercontent.com/3853003/120720707-74f85d00-c4c4-11eb-96c8-e383042f57c5.mov

#### iOS

https://user-images.githubusercontent.com/3853003/120720189-93118d80-c4c3-11eb-9a70-5206013e556d.mov

#### Android

https://user-images.githubusercontent.com/3853003/120720192-960c7e00-c4c3-11eb-8131-18bbe14e937e.mov
